api/envoy/config/cluster/redis/redis_cluster.proto

```diff
@@ -58,11 +59,21 @@ message RedisClusterConfig {
   google.protobuf.Duration cluster_refresh_timeout = 2 [(validate.rules).duration = {gt {}}];
 
   // The minimum interval that must pass after triggering a topology refresh request before a new
-  // request can possibly be triggered again. Any redirection errors received during one of these
+  // request can possibly be triggered again. Any errors received during one of these
   // time intervals are ignored. If not set, this defaults to 5s.
   google.protobuf.Duration redirect_refresh_interval = 3;
 
   // The number of redirection errors that must be received before
   // triggering a topology refresh request. If not set, this defaults to 5.
+  // If this is set to 0, topology refresh after redirect is disabled.
   google.protobuf.UInt32Value redirect_refresh_threshold = 4;
+
+  // The number of failures that must be received before triggering a topology refresh request.
+  // If not set, this defaults to 0, which disables the topology refresh due to failure.
+  google.protobuf.UInt32Value failure_refresh_threshold = 5;
```

Comment:
This and the below field should just be regular `uint32` since they default to 0.